odoo/tools/safe_eval.py

```diff
@@ -376,77 +1524,29 @@ def safe_eval(expr, globals_dict=None, locals_dict=None, mode="eval", nocopy=Fal
     except ZeroDivisionError:
         raise
     except Exception as e:
-        raise ValueError('%s: "%s" while evaluating\n%r' % (ustr(type(e)), ustr(e), expr))
-def test_python_expr(expr, mode="eval"):
-    try:
-        test_expr(expr, _SAFE_OPCODES, mode=mode)
-    except (SyntaxError, TypeError, ValueError) as err:
-        if len(err.args) >= 2 and len(err.args[1]) >= 4:
-            error = {
-                'message': err.args[0],
-                'filename': err.args[1][0],
-                'lineno': err.args[1][1],
-                'offset': err.args[1][2],
-                'error_line': err.args[1][3],
-            }
-            msg = "%s : %s at line %d\n%s" % (type(err).__name__, error['message'], error['lineno'], error['error_line'])
-        else:
-            msg = ustr(err)
-        return msg
-    return False
+        raise ValueError(f"{type(e)} : {e} while evaluating\n{expr}")
 
+    globals_dict.update(globals_dict_eval)
+    sanitize_context(globals_dict, type_checker)
```

Comment:
we can sanitize globals_dict_eval before doing the update on globals_dict, seems more secure
same bellow

Reply:
The result should be the same :thinking:

Reply:
no really if you have an object with an update method